### PR TITLE
Optimized edge edge sat check

### PR DESCRIPTION
**Pull request by Developer-The-Great**

### CODE REVIEW

File: args/physics/HalfEdgeEdge.h [hunk 1]
```diff
@@ -26,6 +26,16 @@ namespace args::physics
...
+
+		math::vec3 getLocalNormal() const
+		{
+			return face->normal;
+		}
+
+		math::vec3 getLocalEdgeDirection()  const
+		{
+			return *nextEdge->edgePositionPtr - *edgePositionPtr;
+		}
```
Comment (Algo-ryth-mix) [inline args/physics/HalfEdgeEdge.h]: These should receive documentation, also the filename should be `halfedgeedge.hpp` or simmilar

File: args/physics/PhysicsStatics.h [hunk 2]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+        * @param convexA the reference collider 
+        * @param convexB the incident collider
+        * @param transformA the transform of convexA
+        * @param transformB the transform of convexB
+        * @param refEdge [out] the resulting reference Edge
+        * @param incEdge [out] the resulting incident Edge
+        * @param seperatingAxisFound [out] the resulting seperating axis found
+        * @param seperation [out] the amount of seperation
+        * @return returns true if a seperating axis was found
+        */
```
Comment (Algo-ryth-mix) [inline args/physics/PhysicsStatics.h]: +1 indent please

Comment (Algo-ryth-mix) [reply args/physics/PhysicsStatics.h]:
```suggestion
       /** @brief Given 2 ConvexColliders, Goes through every single possible edge combination in order to check for a seperating axis. This is done
         * by creating a minkowski face with each edge combination.
         * @param convexA the reference collider 
         * @param convexB the incident collider
         * @param transformA the transform of convexA
         * @param transformB the transform of convexB
         * @param refEdge [out] the resulting reference Edge
         * @param incEdge [out] the resulting incident Edge
         * @param seperatingAxisFound [out] the resulting seperating axis found
         * @param seperation [out] the amount of seperation
         * @return returns true if a seperating axis was found
         */
```

File: args/physics/PhysicsStatics.h [hunk 3]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+
+                            }
+
+                        }
+                    }
+
+                }
+            }
+         
+
```
Comment (Algo-ryth-mix) [inline args/physics/PhysicsStatics.h]: consider... less spaces ?

File: args/physics/PhysicsStatics.h [hunk 4]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+        */
+        static bool isMinkowskiFace(const math::vec3 transformedA1,const math::vec3 transformedA2,
+            const math::vec3 transformedB1,const math::vec3 transformedB2)
+        {
+            //------------------------ Check if normals created by arcA seperate normals of B --------------------------------------//
+
+            math::vec3 planeANormal = math::cross(transformedA1, transformedA2);
+
+            float b1DotPlaneAMultB2DotPlaneA = 
+                math::dot(planeANormal, transformedB1) * math::dot(planeANormal, transformedB2);
```
Comment (Algo-ryth-mix) [inline args/physics/PhysicsStatics.h]: b1DotPlaneAMultB2DotPlaneA... while very descriptive is a bit over the top

File: args/physics/PhysicsStatics.h [hunk 5]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+        * @return returns true if the given arcs intersect
+        */
+        static bool isMinkowskiFace(const math::vec3 transformedA1,const math::vec3 transformedA2,
+            const math::vec3 transformedB1,const math::vec3 transformedB2)
+        {
+            //------------------------ Check if normals created by arcA seperate normals of B --------------------------------------//
+
+            math::vec3 planeANormal = math::cross(transformedA1, transformedA2);
+
+            float b1DotPlaneAMultB2DotPlaneA =
```
Comment (tuhri444) [inline args/physics/PhysicsStatics.h]: Please write a name and not a cryptic sentence.

File: args/physics/PhysicsStatics.h [hunk 6]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+            {
+                return false;
+            }
+
+            //------------------------ Check if normals created by arcB seperate normals of A --------------------------------------//
+
+            math::vec3 planeBNormal = math::cross(transformedB1, transformedB2);
+
+            float a1DotPlaneBMultA2DotPlaneB =
+                math::dot(planeBNormal, transformedA1) * math::dot(planeBNormal, transformedA2);
```
Comment (Algo-ryth-mix) [inline args/physics/PhysicsStatics.h]: same here

File: args/physics/PhysicsStatics.h [hunk 7]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+            if (b1DotPlaneAMultB2DotPlaneA > 0.0f ||  math::epsilonEqual(b1DotPlaneAMultB2DotPlaneA,0.0f,math::epsilon<float>()))
+            {
+                return false;
+            }
+
+            //------------------------ Check if normals created by arcB seperate normals of A --------------------------------------//
+
+            math::vec3 planeBNormal = math::cross(transformedB1, transformedB2);
+
+            float a1DotPlaneBMultA2DotPlaneB =
```
Comment (tuhri444) [inline args/physics/PhysicsStatics.h]: Same shit again, please make it readible.

File: args/physics/PhysicsStatics.h [hunk 8]
```diff
@@ -79,5 +84,180 @@ namespace args::physics
...
+            {
+                return false;
+            }
+
+            //------------------------ Check if arcA and arcB are in the same hemisphere --------------------------------------------//
+
+            math::vec3 abNormal = math::cross(transformedA2, transformedB2);
+
+            float a1DotAbNormalMultb1DotAbNormal = math::dot(abNormal, transformedA1) * math::dot(abNormal, transformedB1);
+
```
Comment (Algo-ryth-mix) [inline args/physics/PhysicsStatics.h]: yes

**Review by Algo-ryth-mix — APPROVED** [review]

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]
wrong button sorry

File: args/physics/colliders/convexcollider.cpp [hunk 9]
```diff
@@ -27,6 +27,18 @@ namespace args::physics
             return;
         }
 
+        HalfEdgeEdge* edgeRef = nullptr;
+        HalfEdgeEdge* edgeInc = nullptr;
+        math::vec3 edgeNormal;
+        float AtoBEdgeSeperation;
```
Comment (tuhri444) [inline args/physics/colliders/convexcollider.cpp]: camelCase

**Review by tuhri444 — CHANGES_REQUESTED** [review]
HalfEdgeEdge.h should have only lowercase letters

**Review by Algo-ryth-mix — APPROVED** [review]

**Review by tuhri444 — APPROVED** [review]

**Pull request merged**